test/jdk/java/net/httpclient/ShortResponseBody.java

```diff
@@ -119,12 +122,17 @@ static String name(ITestResult result) {
 
     @BeforeMethod
     void beforeMethod(ITestContext context) {
-        System.gc();
-        try {
-            Thread.sleep(PAUSE_FOR_GC);
-        } catch (InterruptedException x) {
+        if (client == null || numberOfRequests == REQUESTS_PER_CLIENT) {
+            out.println("--- new client");
+            client = newHttpClient();
+            System.gc();
+            try {
+                Thread.sleep(PAUSE_FOR_GC);
+            } catch (InterruptedException x) {
 
+            }
         }
+        numberOfRequests++;
```

Comment:
Did you mean `numberOfRequests = (numberOfRequests % REQUESTS_PER_CLIENT) + 1;` to match the description of the changes?

Reply:
Good catch! Forgot to reset the counter. Fix pushed.